pkg/pipelines/yaml/resources_test.go

```diff
@@ -0,0 +1,56 @@
+package yaml
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/redhat-developer/kam/pkg/pipelines/namespaces"
+	res "github.com/redhat-developer/kam/pkg/pipelines/resources"
+	"github.com/spf13/afero"
+	"sigs.k8s.io/yaml"
+)
+
+func TestWriteResources(t *testing.T) {
+	fs := afero.NewOsFs()
+	homeEnv := "HOME"
+	originalHome := os.Getenv(homeEnv)
+	defer os.Setenv(homeEnv, originalHome)
+	path, cleanup := makeTempDir(t)
+	defer cleanup()
+	os.Setenv("HOME", path)
+	sampleYAML := namespaces.Create("test", "https://github.com/org/test")
+	r := res.Resources{
+		"test/myfile.yaml": sampleYAML,
+	}
+
+	_, err := WriteResources(fs, "~/manifest", r)
+	assertNoError(t, err)
+
+	want, err := yaml.Marshal(sampleYAML)
+	assertNoError(t, err)
+	got, err := ioutil.ReadFile(filepath.Join(path, "manifest/test/myfile.yaml"))
+	assertNoError(t, err)
+	if diff := cmp.Diff(want, got); diff != "" {
+		t.Fatalf("files not written to correct location: %s", diff)
+	}
+}
+
```

Comment:
Should we call our ol friend afero here?
fakeFs := ioutils.NewMemoryFilesystem()
path := afero.GetTempDir(fakeFs, "test")

Reply:
Can you explain a bit more about this?

Are you proposing to mock out `path/filepath` ?

Reply:
@wtam2018 `homedir.Expand` in WriteResources() doesn't work with afero's in-memory filesystem. Hence the underlying fs is being used